File: Dataset/Leetcode/valid/125/529.java
 class Solution {
    public boolean XXX(String s) {
        int n = s.length();
        if(n <= 1) return true;
        int left = 0;
        int right = n-1;
        char ch[] = s.toLowerCase().toCharArray();
        while(left <= right){
            if(!((ch[left]<='9' && ch[left] >= '0') || (ch[left]<='z' && ch[left] >= 'a'))){
                left++;
                continue;
            }
            if(!((ch[right]<='9' && ch[right] >= '0') || (ch[right]<='z' && ch[right] >= 'a'))){
                right--;
                continue;
            }
            if(ch[left] != ch[right]) return false;
            left++;
            right--;
        }
        return true;
    }
}

File: Dataset/Leetcode/test/27/529.py
 class Solution:
    def XXX(self, nums: List[int], val: int) -> int:
        if len(nums) == 0:
            return 0
        left = right = s = 0
        for i in range(len(nums)):
            if nums[right] != val:
                nums[left],nums[right] = nums[right],nums[left]
                left += 1
            right += 1
        return left

